boa/environment.py

```diff
@@ -344,7 +345,12 @@ def apply_computation(cls, state, msg, tx_ctx):
         # print("ENTER", Address(msg.code_address or bytes([0]*20)), contract)
         if contract is None or not cls.env._fast_mode_enabled:
             # print("SLOW MODE")
-            return super().apply_computation(state, msg, tx_ctx)
+            computation = super().apply_computation(state, msg, tx_ctx)
+            if computation.is_error:
+                # After the computation is applied with an error the state is reverted
+                # Before the revert, save the contract representation for the error message
+                computation._contract_repr_before_revert = repr(contract)
```

Comment:
Alternatively we could extract more data at this point (e.g. storage.dump())

Reply:
let's pull the whole storage dump for now. it seems like it could come in handy

Reply:
never mind -- i agree with your approach of getting the whole contract repr